src/page/mod.rs

```diff
@@ -63,6 +63,27 @@ pub enum CacheEntry<M> {
 
 unsafe impl<M: Send> Send for CacheEntry<M> {}
 
+#[derive(Debug, Clone)]
+pub struct CasKey<P> {
+    ptr: *const stack::Node<CacheEntry<P>>,
+    tag: usize,
+}
+
+impl<'s, P> From<Ptr<'s, stack::Node<CacheEntry<P>>>> for CasKey<P> {
+    fn from(ptr: Ptr<'s, stack::Node<CacheEntry<P>>>) -> CasKey<P> {
+        CasKey {
+            ptr: ptr.as_raw(),
+            tag: ptr.tag(),
+        }
+    }
+}
+
+impl<'s, P> Into<Ptr<'s, stack::Node<CacheEntry<P>>>> for CasKey<P> {
+    fn into(self) -> Ptr<'s, stack::Node<CacheEntry<P>>> {
+        unsafe { Ptr::from_raw(self.ptr).with_tag(self.tag) }
+    }
+}
```

Comment:
this lets us convert `CasKey`'s to coco `Ptr`'s so that we can pass the pointer out of the coco-scope protected code, and be used later for attempting to compare-and-swap a coco-protected variable.